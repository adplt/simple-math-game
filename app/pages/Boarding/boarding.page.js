import React, {Component} from 'react';
import {
  // Platform,
  Text,
  View,
} from 'react-native';
import styles from './boarding.page.style';
import SplashScreen from 'react-native-splash-screen';
import Touchable from '../../components/Touchable/Touchable.component';
import PropTypes from 'prop-types';

export default class Boarding extends Component {

  static propTypes = {
    navigation: PropTypes.object,
  }

  componentDidMount () {
    SplashScreen.hide();
  }

  onPress = (params) => {
    const {navigation} = this.props;
    navigation.navigate('Level', params);
  }

  goToCounting = () => this.onPress({menu: 'count'});

  goToChoosing = () => this.onPress({menu: 'choose'});

  render () {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          {'Mathematic'}
        </Text>
        <Touchable onPress={this.goToCounting} style={{padding: 10}}>
          <Text style={styles.instructions}>
            {'Counting'}
          </Text>
        </Touchable>
        <Touchable onPress={this.goToChoosing} style={{padding: 10}}>
          <Text style={styles.instructions}>
            {'Choosing'}
          </Text>
        </Touchable>
      </View>
    );
  }
}
