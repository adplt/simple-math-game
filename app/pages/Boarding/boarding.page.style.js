export default {
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#EEEEEE',
  },
  welcome: {
    fontSize: 50,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
  menuTouchable: {
    padding: 10
  }
};
