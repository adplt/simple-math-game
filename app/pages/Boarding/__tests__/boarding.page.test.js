import React from 'react';
import renderer from 'react-test-renderer';
import Boarding from '../boarding.page';

describe('ScrollableViewOverlay component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Boarding />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
