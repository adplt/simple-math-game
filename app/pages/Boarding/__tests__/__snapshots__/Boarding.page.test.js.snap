// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Boarding component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EEEEEE",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 50,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Mathematic
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "padding": 10,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#333333",
          "marginBottom": 5,
          "textAlign": "center",
        }
      }
    >
      Counting
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "padding": 10,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#333333",
          "marginBottom": 5,
          "textAlign": "center",
        }
      }
    >
      Choosing
    </Text>
  </View>
</View>
`;
